varname.py

```diff
@@ -254,7 +261,35 @@ def nameof(*args):
         raise VarnameRetrievingError("At least one variable should be "
                                      "passed to nameof")
 
-    return ret[0] if len(args) == 1 else tuple(ret)
+    if len(args) == 1:
+        ret = ret[0]
+        if TESTING:
```

Comment:
Can we get rid of this `TESTING`, and construct something like `_bytecode_nameof()` in the tests?

Reply:
I don't understand your alternative. `_bytecode_nameof` is already used in non-testing situations just shortly above. This is just an easy way to test both implementations at the same time. To do it without `if` we could either construct an alternative `nameof` in the tests which calls both like this, or do a lot of pytest parametrizing.

Reply:
I just don't want to mix testing stuff in the source code. What about we compose a `nameof` function in the test but use `_bytecode_nameof` instead of normal routing, and test if the names from bytecode and normal routing are the same?

The `nameof` function in the test would be very complicated, say we can just do:
```python
def test_bytenameof_sameas_normal():
  def byte_nameof(arg): # we can just test len(args) == 1
    return varname_module._bytecode_nameof()
  
  a = {}
  assert byte_nameof(a) == varname_module.nameof(a)
```

Reply:
I don't want the bytecode version to just be tested once, I want to make sure that it passes all the different kinds of tests you've put the usual varname through.

Reply:
Do you mean to replace the normal routing with it finally?

Reply:
No keep the normal route, I trust it more and only it can support multiple arguments. But test both fully since the bytecode version will still be used sometimes as a fallback.

Reply:
If it is only used as a fallback, then why don't just test the fallback cases?

Reply:
I want to make use of basically all the testing logic from lines 253 to 296: test_nameof, test_nameof_statements, test_nameof_expr.

Reply:
I see. What about making the bytecode `nameof` as a fixture? I don't mind a duplicate of lines 253-296.